File: src/org/cappuccino_project/ide/intellij/plugin/stubs/types/ObjJIncludeFileStubType.java
package org.cappuccino_project.ide.intellij.plugin.stubs.types;

import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJIncludeFileImpl;
import org.cappuccino_project.ide.intellij.plugin.stubs.interfaces.ObjJImportStub;
import org.jetbrains.annotations.NotNull;

public class ObjJIncludeFileStubType extends ObjJImportStatementStubType<ObjJIncludeFileImpl> {

    public ObjJIncludeFileStubType(
            @NotNull
                    String debugName) {
        super(debugName, ObjJIncludeFileImpl.class);
    }

    @Override
    public ObjJIncludeFileImpl createPsi(
            @NotNull
                    ObjJImportStub<ObjJIncludeFileImpl> stub) {
        return new ObjJIncludeFileImpl(stub, this);
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/stubs/types/ObjJIncludeFileStubType.kt
package org.cappuccino_project.ide.intellij.plugin.stubs.types

import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJIncludeFileImpl
import org.cappuccino_project.ide.intellij.plugin.stubs.interfaces.ObjJImportStub

class ObjJIncludeFileStubType(
        debugName: String) : ObjJImportStatementStubType<ObjJIncludeFileImpl>(debugName, ObjJIncludeFileImpl::class.java) {

    override fun createPsi(
            stub: ObjJImportStub<ObjJIncludeFileImpl>): ObjJIncludeFileImpl {
        return ObjJIncludeFileImpl(stub, this)
    }
}
